util/contentutil/multiprovider.go

```diff
@@ -0,0 +1,43 @@
+package contentutil
+
+import (
+	"context"
+	"sync"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/errdefs"
+	digest "github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+)
+
+func NewMultiProvider(base content.Provider) *MultiProvider {
```

Comment:
I wonder this can be simplified as follows for most usecases:
```go
func NewMultiProvider(providers ...content.Provider) *MultiProvider
...
func (mp *MultiProvider) ReaderAt(ctx context.Context, dgst digest.Digest) (content.ReaderAt, error) {
  for _, provider := range mp.providers{
  ...
  }
  ...
}

Reply:
That is a bit different. Multiprovider doesn't search through all the providers to find the record but a provider must be connected to a specific record with `add`. And the base is just a fallback. The problem with just scanning through all the providers is that `ReadetAt` can be a long-running function and maybe call to network, so we don't want to make these calls that  don't return anything.